src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx.cs

```diff
@@ -523,6 +539,11 @@ public static Vector256<uint> Insert(Vector256<uint> value, uint data, byte inde
         /// </summary>
         public static Vector256<long> Insert(Vector256<long> value, long data, byte index)
         {
+            if (IntPtr.Size != 8)
+            {
+                throw new PlatformNotSupportedException();
+            }
+
```

Comment:
Same as with `Extract`, we should determine if this is "better" than what all three native compilers do, which is:
```asm
vpinsrb
vinsertf128
```